### PR TITLE
Harness with libfuzzer

### PR DESCRIPTION
**Pull request by jeff14994**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.9)
```
Comment (xansec) [inline CMakeLists.txt]: Did you add a new `CMakeLists.txt` or move the existing one to a new location?

File: fuzz.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#include <libriscv/machine.hpp>
```
Comment (xansec) [inline fuzz.cpp]: Was this written from scratch or copied from somewhere? `fuzz.cpp` is not a very descriptive filename.

File: helpers.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#include <climits>
```
Comment (xansec) [inline helpers.cpp]: This file also seems to have been moved?

**Review by xansec — CHANGES_REQUESTED** [review]
It looks like you've taken the repository and copied it in its entirety underneath the `/fuzz` directory. This won't work, will unnecessarily duplicate code, and confuse the upstream maintainers. I would recommend trying to start again with this target, adding only one or a couple of files. The bare minimum you should need is

1. A Dockerfile
2. A Mayhemfile
3. A mayhem.yml (for github/workflows)
4. A fuzz target 

The bare minimum you should have to do is described here in this subfolder:
https://github.com/fwsGonzo/libriscv/tree/master/fuzz

If you see there, the `fuzzer.sh` builds the target for you. All you need to do is run this command in your Dockerfile (during the build stage), and then copy the relevant fuzz target (or targets) to the package stage. Finally, create one or more Mayhemfiles referencing each fuzz target.

I would recommend closing this PR and resubmitting.

**Comment by xansec** [comment]
Please update this PR or we will have to close.

**Comment by jeff14994** [comment]
okay, i'm working on it

**Pull request closed without merging**